src/main/java/com/nkh/ECommerceShop/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,70 @@
+package com.nkh.ECommerceShop.security;
+
+import com.nkh.ECommerceShop.security.jwt.AuthEntryPointJwt;
+import com.nkh.ECommerceShop.security.jwt.AuthTokenFilter;
+import com.nkh.ECommerceShop.security.service.UserDetailsServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableMethodSecurity
+public class WebSecurityConfig {
+    @Autowired
```

Comment:
can you migrate to use constructor autowiring? https://stackoverflow.com/a/40620318

Reply:
done